src/main/java/org/jenkinsci/plugins/environmentdashboard/DeploymentView.java

```diff
@@ -2,60 +2,85 @@
 
 import hudson.Extension;
 import hudson.Util;
-import hudson.model.*;
+import hudson.model.Job;
+import hudson.model.ListView;
+import hudson.model.TopLevelItem;
+import hudson.model.ViewDescriptor;
 import hudson.util.FormValidation;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
 import net.sf.json.JSONObject;
+import org.jenkinsci.plugins.environmentdashboard.Deployment.DeploymentAction;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.jenkinsci.plugins.workflow.multibranch.WorkflowMultiBranchProject;
 import org.kohsuke.stapler.DataBoundConstructor;
 import org.kohsuke.stapler.QueryParameter;
 import org.kohsuke.stapler.StaplerRequest;
 
 import javax.annotation.Nonnull;
-import java.util.*;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
 import java.util.regex.Pattern;
 import java.util.regex.PatternSyntaxException;
+import java.util.stream.Collectors;
 
 public class DeploymentView extends ListView {
     @DataBoundConstructor
     public DeploymentView(String name) {
         super(name);
     }
 
-    public Map<String, Deployment.DeploymentAction> getEnvironments(TopLevelItem item) {
-        if (!(item instanceof Job)) {
-            return new TreeMap<>();
+    private List<Unit.Environment> getEnvs(TopLevelItem item) {
+        List<WorkflowRun> runs = Collections.emptyList();
+        if (item instanceof WorkflowMultiBranchProject) {
+            runs = ((WorkflowMultiBranchProject) item)
+                    .getItems()
+                    .stream()
+                    .map(Job::getBuilds)
+                    .flatMap(Collection::stream)
+                    .collect(Collectors.toList());
+        } else if (item instanceof WorkflowJob) {
+            runs = ((WorkflowJob) item).getBuilds();
         }
```

Comment:
Shouldn't we return empty list for unexpected types of jobs?

Reply:
We do that. The first line initializes an empty List<WorkflowRun> and then we convert it to the empty List<Environment>. Then we filtered jobs with empty environments at all in the getUnits() method

Reply:
you're right, sorry, missed that line...